x-pack/test/observability_solution_api_integration/test_suites/alerting/burn_rate/burn_rate_rule.ts

```diff
@@ -23,8 +17,11 @@ export default function ({ getService }: FtrProviderContext) {
   const alertingApi = getService('alertingApi');
   const dataViewApi = getService('dataViewApi');
   const sloApi = getService('sloApi');
+  const config = getService('config');
+  const isServerless = config.get('serverless');
+  const expectedConsumer = isServerless ? 'observability' : 'slo';
```

Comment:
Let me elaborate a little bit more on this. In our rules we have a `consumer` field, which on a serverless environment gets the value `observability` where as in a stateful environment it gets the value of the app, for example `slo` in this case. So based on the environment we expect a different value for the consumer field.

In this [PR](https://github.com/elastic/kibana/pull/162698) there was following comment: 

`It is extremely forbidden to add conditionals on assertions or actions to make a test work in both environments. That is a red flag warning you about having to create a specific test for it.`

Based on above comment, we should have a separate test for this. I could move into the `test_serverless` folder only the test that checks the consumer field, that's no problem. But I would need to move a few more things to make my test work (for example the creation of the rule and whatever happens in the beforeEach), which eventually means that tests inside the new observability_solution_api_tests would be very similar with the test_serverless tests apart from a few different test cases.

 - **observability_solution_api_tests**: create a rule, check if rule is active, check information in the alert document, check if consumer is slo
 - **test_serverless**:  create a rule, check if consumer is observability`

I would like your thoughts on this.

Reply:
In general, it's not advisable to have these kind of code branches. The need for such branches is an indicator that stateful and serverless are handling things differently and thus probably warrants separate tests. However, there might be exception cases where a small code branch is used to avoid big overhead.